project/profiles/capacity4more/modules/c4m/content/c4m_content_photoalbum/c4m_content_photoalbum.module

```diff
@@ -23,6 +23,39 @@ function c4m_content_photoalbum_c4m_og_vocab_info_content() {
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Adds an `Add Photos` button to the `Create Photo album` form.
+ */
+function c4m_content_photoalbum_form_photoalbum_node_form_alter(&$form, &$form_instance, $form_id) {
+  $form['actions']['add_photos'] = array(
+    '#type' => 'submit',
+    '#value' => t('Add photos'),
+    '#weight' => 7,
+    '#submit' => array('c4m_content_photoalbum_add_photos_submit'),
+  );
+}
+
+/**
+ * Form submission handler for `Create Photo album`.
+ *
+ * Handles the 'Add photos' button on the Create Photo album form.
+ * Clicking this button publishes the photo album and redirects to the
+ * 'Upload New Items' tab.
+ */
+function c4m_content_photoalbum_add_photos_submit($form, &$form_state) {
+  // Publish the form by calling the default function.
+  node_form_submit($form, $form_state);
```

Comment:
this seems wrong, this submit handler should just be called after the original submit handler was called. and then you can set `$form_state['redirect'] = format_string('node/' . $form_state['nid'] . '/upload'); (i.e. no need for format_string - which should be used only when outputting user generated string)